### DISCUSSION TITLE
How how expand JSON array items result?

### DISCUSSION THREAD

**Discussion by dengyakui** [datalust/seq-tickets#1938] [Help] [answered]
We had record  array in logs,  below is the stucture, it is a JSON array string, every item is a git commit's line changed info :
```JSON
[ 
 { "commitId": "<guid>",  "author": "<author>", "deleteLine": 10, "addLine": 20 }  ,
 { "commitId": "<guid>",  "author": "<author>", "deleteLine": 11, "addLine": 20 }  ,
 { "commitId": "<guid>",  "author": "<author>", "deleteLine": 12, "addLine": 20 }  ,
 { "commitId": "<guid>",  "author": "<author>", "deleteLine": 13, "addLine": 20 }  ,
]
```
Now, we have a problem for query result, below is our query :
![image](https://github.com/datalust/seq-tickets/assets/3945919/850a50bc-2431-4112-94e1-765b52cb49a2)

The query result's every record has an array obj,  and we want to directly expand the array's all items and only select every item's some property, like `SelectMany(r=>r.items).select(item=>new{ item.addLine, item.deleteLine })` in LINQ syntax. 
We have tried to use array[*].addLine and  array[?].deleteLine syntax,  but it only works for where clause, can not used for select.

![image](https://github.com/datalust/seq-tickets/assets/3945919/662cf202-ebd7-4070-a054-e264ceded604)

Wondering is there some func or operator for us to expand collection  items and do some aggregation func, like `Sum( arr[*].addLine )`, `Avg(arr[*].addLine)`

**Comment by liammclennan** [comment, marked as answer]
Hi @dengyakui ,

If you can store structured data (instead of a string of JSON) then you can use `lateral unnest` to flatten the collection. See https://blog.datalust.co/exploding-collections/.

**Reply by dengyakui** [reply]
@liammclennan :
Thanks,   that is just what i want .
By the way , I had read a lot of the online docs before,  but not found the newly `lateral unnest` , May be the doc should be updated sometime.
Thank agagin.

**Reply by liammclennan** [reply]
That's great. The doc link is https://docs.datalust.co/docs/the-seq-query-language#lateral-cross-joins
